Marlin/src/inc/SanityCheck.h

```diff
@@ -372,27 +372,45 @@ static_assert(X_MAX_LENGTH >= X_BED_SIZE && Y_MAX_LENGTH >= Y_BED_SIZE,
  * Filament Runout needs a pin and either SD Support or Auto print start detection
  */
 #if ENABLED(FILAMENT_RUNOUT_SENSOR)
-  #if !HAS_FIL_RUNOUT
-    #error "FILAMENT_RUNOUT_SENSOR requires FIL_RUNOUT_PIN."
+  #error "FILAMENT_RUNOUT_SENSOR is now FILAMENT_RUNOUT_MULTI_SENSORS. Please update your configuration."  
+#endif   
+#if ENABLED(FILAMENT_RUNOUT_MULTI_SENSORS)
+  #if FIL_RUNOUT_SENSORS >5 || FIL_RUNOUT_SENSORS <1
+    #error "FILAMENT_RUNOUT_MULTI_SENSORS Number of sensors error"
+  #elif  FIL_RUNOUT_SENSORS >0 && !HAS_FIL_RUNOUT0
+    #error "FILAMENT_RUNOUT_MULTI_SENSORS requires FIL_RUNOUT_0_PIN"
+  #elif  FIL_RUNOUT_SENSORS >1 && (!HAS_FIL_RUNOUT0 || !HAS_FIL_RUNOUT1)
+    #error "FILAMENT_RUNOUT_MULTI_SENSORS requires FIL_RUNOUT_PIN 0 to 1"
+  #elif  FIL_RUNOUT_SENSORS >2 && (!HAS_FIL_RUNOUT0 || !HAS_FIL_RUNOUT1 || !HAS_FIL_RUNOUT2)
+    #error "FILAMENT_RUNOUT_MULTI_SENSORS requires FIL_RUNOUT_PIN 0 to 2"
+  #elif  FIL_RUNOUT_SENSORS >3 && (!HAS_FIL_RUNOUT0 || !HAS_FIL_RUNOUT1 || !HAS_FIL_RUNOUT2 || !HAS_FIL_RUNOUT3)
+    #error "FILAMENT_RUNOUT_MULTI_SENSORS requires FIL_RUNOUT_PIN 0 to 3"
+  #elif  FIL_RUNOUT_SENSORS >4 && (!HAS_FIL_RUNOUT0 || !HAS_FIL_RUNOUT1 || !HAS_FIL_RUNOUT2 || !HAS_FIL_RUNOUT3 || !HAS_FIL_RUNOUT4)
+    #error "FILAMENT_RUNOUT_MULTI_SENSORS requires FIL_RUNOUT_PIN 0 to 4"
```

Comment:
i have writtend these rules like this 
To give more informatin on compilation errors for the user

Reply:
Many safety need to be added , but i have placed the minimum 
I don't understand the coding it's very hard
it's FOR YOU
lol